packages/navigator/src/browser/navigator-model.ts

```diff
@@ -37,6 +50,49 @@ export class FileNavigatorModel extends FileTreeModel {
         }
     }
 
+    async updateRoot(): Promise<void> {
+        this.root = await this.createRoot();
+    }
+
+    protected async createRoot(): Promise<TreeNode | undefined> {
+        const workspaceDescription = await this.workspaceService.workspace;
+        const roots = workspaceDescription.roots;
+        if (roots.length > 0) {
+            const children: DirNode[] = [];
+            for (const r of roots) {
+                const icon = await this.labelProvider.getIcon(r);
+                children.push({
+                    id: r.uri,
+                    uri: new URI(r.uri),
+                    name: this.labelProvider.getName(new URI(r.uri)),
+                    icon,
+                    visible: true,
+                    parent: {
```

Comment:
You should create a workspace node first, then its children with a proper parent and then assign children to a workspace node. I would add `NavigatorTree.createWorkspaceRoot` method which accepts `FileStat` and `WorkspaceNode` and delegates to `super.toNode(fileStat, parent)` to create a workspace root node, then this code can be rewritten as:
```ts
        const workspace = WorkspaceNode.create();
        workspace.children = workspaceDescription.roots.map(root =>
            this.tree.createWorkspaceRoot(root, workspace)
        );
        return workspace;
```